notebooks/02-ingestion/10-updating-data.ipynb

```diff
@@ -0,0 +1,1188 @@
+    "   \"origin\", \n",
+    "   \"destination\", \n",
+    "   \"departure_delay\"\n",
+    "FROM \"example-flights-updates\" \n",
+    "WHERE TIME_IN_INTERVAL(\"__time\", '2005-11-01T00:15:00.000Z/PT1S')\n",
+    "  AND NOT ( \"__time\"= '2005-11-01 00:15:00.000Z' \n",
+    "                AND \"airline\"='AA' \n",
+    "                AND \"flight_number\"=513 )\n",
+    "PARTITIONED BY FLOOR(__time TO SECOND) \n",
+    "'''\n",
+    "display.run_task(sql)\n",
+    "sql_client.wait_until_ready('example-flights-updates')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "01c04dcc-45c7-49c7-bae7-314bd900b12e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# how did the segments change for the 1 day timeframe for 11/01/2005 <= t < 11/02/2005\n",
+    "sql='''\n",
+    "SELECT \"start\", \"end\", \"num_rows\", \"version\" \n",
+    "FROM sys.segments \n",
+    "WHERE datasource='example-flights-updates'\n",
+    "  AND \"start\" >=  '2005-11-01T00:00:00.000Z' AND \"start\" < '2005-11-02T00:00:00.000Z'\n",
+    "'''\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b668f50e-9817-4a3b-b6c6-36fc533c591d",
+   "metadata": {},
+   "source": [
+    "The 1-second segment was overshadowed by the new 1-second segment, examine the version you see here and compare it to the version in the prior 1-second segment shown earlier in the notebook. The version is a newer timestamp and the new segment completely covers the time-frame of the prior segment. Since the prior segment is overshadowed and no longer needed, Druid removes it.\n",
+    "\n",
+    "Notice that the new 1-second segment only contains one row, if you query the checksums now, you can see the change:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2f725acc-ba37-4e49-b0e1-b9f156bd4087",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT count(*) \"total_flights_day\", SUM(\"departure_delay\") \"total_time_lost\" \n",
+    "FROM \"example-flights-updates\"\n",
+    "WHERE TIME_IN_INTERVAL(\"__time\", '2005-11-01T00:00:00.000Z/P1D')  -- totals for the day\n",
+    "'''\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fce16895-ecac-45f0-b643-8b90fe9f5766",
+   "metadata": {},
+   "source": [
+    "## Replacing a complete time interval with new data \n",
+    "Sometimes a full set of the data is delivered as an updated batch file that covers a known timeframe. For example, a whole day.\n",
+    "\n",
+    "Since the flight sample data we loaded does not come with a change set, the following cell simulates one by writing a day's worth of flights to another table while introducing a change to the `departure_delay` values. The generated table only includes the data for the day `2005-11-02` as expressed in the following SQL `WHERE TIME_IN_INTERVAL(\"__time\", '2005-11-02T00:00:00.000Z/P1D')`:"
```

Comment:
I'm using "change set" as a more generic term meaning any set of changes to the data. In one case it is a delta of a few existing rows + some new rows, in case it is a whole day of data and in another case it is a whole month but for just a single airline which removes existing flights for that airline and replaces them with a new set of flights throughout the timeline.